frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.boot-order.scenario.ts

```diff
@@ -0,0 +1,137 @@
+import { getBootableDevices, getBootableDevicesInOrder } from '../../src/selectors/vm/devices';
+import { VM_CREATE_AND_EDIT_TIMEOUT_SECS } from './utils/consts';
+import { VirtualMachine } from './models/virtualMachine';
+import { getVMManifest } from './utils/mocks';
+import { getRandStr, getResourceObject, getSelectOptions, selectOptionByText } from './utils/utils';
+
+describe('KubeVirt VM detail - Boot Order Dialog', () => {
+  const JS_HTML5_DND =
+    'function e(e,t,n,i){var r=a.createEvent("DragEvent");r.initMouseEvent(t,!0,!0,o,0,0,0,c,g,!1,!1,!1,!1,0,null),Object.defineProperty(r,"dataTransfer",{get:function(){return d}}),e.dispatchEvent(r),o.setTimeout(i,n)}var t=arguments[0],n=arguments[1],i=arguments[2]||0,r=arguments[3]||0;if(!t.draggable)throw new Error("Source element is not draggable.");var a=t.ownerDocument,o=a.defaultView,l=t.getBoundingClientRect(),u=n?n.getBoundingClientRect():l,c=l.left+(l.width>>1),g=l.top+(l.height>>1),s=u.left+(u.width>>1)+i,f=u.top+(u.height>>1)+r,d=Object.create(Object.prototype,{_items:{value:{}},effectAllowed:{value:"all",writable:!0},dropEffect:{value:"move",writable:!0},files:{get:function(){return this._items.Files}},types:{get:function(){return Object.keys(this._items)}},setData:{value:function(e,t){this._items[e]=t}},getData:{value:function(e){return this._items[e]}},clearData:{value:function(e){delete this._items[e]}},setDragImage:{value:function(e){}}});if(n=a.elementFromPoint(s,f),!n)throw new Error("The target element is not interactable and need to be scrolled into the view.");u=n.getBoundingClientRect(),e(t,"dragstart",101,function(){var i=n.getBoundingClientRect();c=i.left+s-u.left,g=i.top+f-u.top,e(n,"dragenter",1,function(){e(n,"dragover",101,function(){n=a.elementFromPoint(c,g),e(n,"drop",1,function(){e(t,"dragend",1,callback)})})})})';
+  const testVM = getVMManifest('Container', testName, `bootordervm-${getRandStr(5)}`);
+  const vm = new VirtualMachine(testVM.metadata);
+
+  beforeAll(() => {
+    createResource(testVM);
+  });
+
+  afterAll(() => {
+    deleteResource(testVM);
+  });
+
+  beforeEach(async () => {
+    await vm.navigateToDetail();
+    await vm.modalEditBootOrder();
+  });
+
+  it(
+    'Displays boot devices',
+    async () => {
+      const bootableDevices = getBootableDevices(getResourceObject(vm.name, vm.namespace, vm.kind));
+      const displayedbootableDevices = await vm.getBootDevices();
+      expect(displayedbootableDevices.length).toEqual(bootableDevices.length);
+    },
+    VM_CREATE_AND_EDIT_TIMEOUT_SECS,
+  );
+
+  it(
+    'Deletes bootable device',
+    async () => {
+      const FIRST_DEVICE_POSITION = 1;
+      const initialBootableDevices = getBootableDevicesInOrder(
+        getResourceObject(vm.name, vm.namespace, vm.kind),
+      );
+      await click(bootOrderView.deleteDeviceButton(FIRST_DEVICE_POSITION));
+      await click(bootOrderView.saveButton);
+      // Wait for the boot Order to update
+      await browser.wait(
```

Comment:
can we make a function for the code bellow till the end of this function to avoid duplication of code?

Reply:
Done